vendor/github.com/manicminer/hamilton/odata/odata.go

```diff
@@ -0,0 +1,124 @@
+package odata
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+)
+
+const (
+	ErrorAddedObjectReferencesAlreadyExist   = "One or more added object references already exist"
+	ErrorConflictingObjectPresentInDirectory = "A conflicting object with one or more of the specified property values is present in the directory"
+	ErrorRemovedObjectReferencesDoNotExist   = "One or more removed object references do not exist"
+)
+
+// OData is used to unmarshall OData metadata from an API response.
+type OData struct {
+	Context      *string `json:"@odata.context"`
+	MetadataEtag *string `json:"@odata.metadataEtag"`
+	Type         *string `json:"@odata.type"`
+	Count        *string `json:"@odata.count"`
+	NextLink     *string `json:"@odata.nextLink"`
+	Delta        *string `json:"@odata.delta"`
+	DeltaLink    *string `json:"@odata.deltaLink"`
+	Id           *string `json:"@odata.id"`
+	Etag         *string `json:"@odata.etag"`
+
+	Error *Error `json:"-"`
+
+	Value *[]json.RawMessage `json:"value"`
+}
+
+func (o *OData) UnmarshalJSON(data []byte) error {
+	// Perform unmarshalling using a local type
+	type odata OData
+	var o2 odata
+	if err := json.Unmarshal(data, &o2); err != nil {
+		return err
+	}
+	*o = OData(o2)
+
+	// Look for errors in the "error" and "odata.error" fields
+	var e map[string]json.RawMessage
+	if err := json.Unmarshal(data, &e); err != nil {
+		return err
+	}
+	for _, k := range []string{"error", "odata.error"} {
```

Comment:
can this also be in `@odata.error`?

Reply:
I haven't seen that but maybe? AFAICT `@odata.error` isn't in the odata standard, although I think i'd probably expect it to have astring value if it showed up anywhere.